src/stdx/sort_test.zig

```diff
@@ -0,0 +1,116 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const stdx = @import("../stdx.zig");
+
+test "sort_stable" {
+    const Value = struct {
+        const Value = @This();
+
+        x: u32, // x determines the order of the values.
+        y: u32, // y ensures that values are distinct for the purpose of checking stability.
+
+        fn compare_x_ascending(_: void, a: Value, b: Value) bool {
+            return a.x < b.x;
+        }
+
+        fn compare_xy_ascending(_: void, a: Value, b: Value) bool {
+            if (a.x < b.x) return true;
+            if (a.x > b.x) return false;
+            return a.y < b.y;
+        }
+
+        fn compare_xy_descending(_: void, a: Value, b: Value) bool {
+            if (a.x > b.x) return true;
+            if (a.x < b.x) return false;
+            return a.y < b.y;
+        }
```

Comment:
Is it intentional that this is descending in x, but ascending in y? The name makes it feel like y should also be descending

Reply:
Ah, I see that this is intentional. Maybe it'd be simpler to set `.y = 0` initially, and then set `.y = @intCast(i)` after we partitioned things?